spec/retry_spec.rb

```diff
@@ -295,4 +295,54 @@ def self.retry_delay(_attempt, _exception)
       end
     end
   end
+
+  describe 'retry callback' do
+    let(:retry_instance) do
+      described_class.new(strategy: :constant, callback: callback, **options)
+    end
+    let(:callback) { proc { raise 'callback executed' } }
```

Comment:
(if the validation is just on responding to `call`, this could be an RSpec `double(call: nil)` here - and then the assertions can be on stuff like `expect(callback).to have_received(:call)`, which feels neater)